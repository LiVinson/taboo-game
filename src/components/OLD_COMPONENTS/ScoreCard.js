import React from "react"

export default function ScoreCard({ score1, score2 }) {
  return (
    <div>
      <p>
        Team 1:<span>{score1}</span>
      </p>
      <p>
        Team 2:<span>{score2}</span>
      </p>
    </div>
  )
}
