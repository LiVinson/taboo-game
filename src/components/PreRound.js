import React from "react";

export default function PreRound({message, isGiver, startRound}) {
    return (
        <React.Fragment>
            <p>{message}</p>
            {isGiver
             ?  <button onClick={startRound}>Start Game</button>
             : null}
        </React.Fragment>
    )
}