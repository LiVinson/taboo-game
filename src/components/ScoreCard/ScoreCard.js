import React from 'react'
import PropTypes from "prop-types"
import { Scores, ScoreTitle, Score } from './style'
import  List  from 'components/shared/List'

const ScoreCard = ({ teamScores }) => {
	return (
		<Scores>
			<ScoreTitle>Scores</ScoreTitle>
			<List>
				{teamScores.map((score, index) => (
					<Score key={index}>
						Team {index + 1}: <span>{score}</span>
					</Score>
				))}
			</List>
		</Scores>
	)
}

ScoreCard.propTypes = {
	teamScores: PropTypes.array.isRequired
}
export default ScoreCard
