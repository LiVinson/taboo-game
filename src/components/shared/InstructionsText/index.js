import { InstructionsText } from "./InstructionsText"

export default InstructionsText