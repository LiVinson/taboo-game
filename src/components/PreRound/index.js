import PreRound from "./PreRound"

export default PreRound