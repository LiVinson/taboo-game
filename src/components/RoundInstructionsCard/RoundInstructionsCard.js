import React from 'react'
import PropTypes from 'prop-types'
import { TabooCard, ButtonTabooCard } from 'components/shared/TabooCard'
import  InstructionsText  from 'components/shared/InstructionsText'
import  KeyWord  from 'components/shared/KeyWord'

//Depending on role of current users, returns a TabooCard with different set of instructions and button
//to start the round
const RoundInstructionsCard = ({ role, giver, watcher, startRound, error }) => {
	let instructions
	//Determine the instructions based on current player's team and role
	switch (role) {
		case 'giver':
			instructions = displayGiverInstructions(watcher, startRound, error)
			break
		case 'watcher':
			instructions = displayWatcherInstructions(giver)
			break
		case 'giverTeam':
			instructions = displayGiverTeamInstructions(giver, watcher)
			break
		case 'watcherTeam':
			instructions = displayWatcherTeamInstructions(giver, watcher)
			break
		default:
			instructions = ''
	}

	return instructions
}

RoundInstructionsCard.propTypes = {
	role: PropTypes.string.isRequired,
	giver: PropTypes.object.isRequired,
	watcher: PropTypes.object.isRequired,
	startRound: PropTypes.func.isRequired,
}

const displayGiverInstructions = (watcher, startRound, error) => {
	const buttonInfo = [
		{
			text: 'Start Round!',
			onClick: startRound,
		},
	]
	return (
		<ButtonTabooCard tabooWord="Get Ready!" buttons={buttonInfo} error={error}>
			<InstructionsText>
				It’s your turn to give clues to your team! {<KeyWord>{watcher.name}</KeyWord>} will be making sure you
				don’t say any taboo words.
			</InstructionsText>

			<InstructionsText>
				Select {<KeyWord>NEXT</KeyWord>} if your team guesses the word correctly or {<KeyWord>SKIP</KeyWord>} to go to the next card. Make sure to keep an
				eye on your time!
			</InstructionsText>
		</ButtonTabooCard>
	)
}

const displayWatcherInstructions = (giver) => {
	return (
		<TabooCard tabooWord="Get Ready!">
			<InstructionsText>
				It’s the other team’s turn to give clues! {<KeyWord>You</KeyWord>} are the watcher for this round! If{' '}
				{<KeyWord>{giver.name}</KeyWord>} says any words from the taboo list or gives any illegal clues, hit the{' '}
				{<KeyWord>buzzer!</KeyWord>}
			</InstructionsText>
		</TabooCard>
	)
}

const displayGiverTeamInstructions = (giver, watcher) => {
	return (
		<TabooCard tabooWord="Get Ready!">
			<InstructionsText>
				It's your turn to guess! {<KeyWord>{giver.name}</KeyWord>} will be giving the clues, and{' '}
				{<KeyWord>{watcher.name}</KeyWord>} will be checking for any illegal clues. Good luck!
			</InstructionsText>
		</TabooCard>
	)
}

const displayWatcherTeamInstructions = (giver, watcher) => {
	return (
		<TabooCard tabooWord="Just Relax!">
			<InstructionsText>
				It's the other team's turn to guess! {<KeyWord>{giver.name}</KeyWord>} will be giving the clues, and{' '}
				{<KeyWord>{watcher.name}</KeyWord>} will be checking for any illegal clues.
			</InstructionsText>
		</TabooCard>
	)
}
export default RoundInstructionsCard
