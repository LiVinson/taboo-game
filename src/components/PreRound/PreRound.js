import React from 'react'
import PropTypes from 'prop-types'
import RoundInstructionsCard from 'components/RoundInstructionsCard'
import ScoreCard from 'components/ScoreCard'
import { TabooCard } from 'components/shared/TabooCard'
import InstructionsText from 'components/shared/InstructionsText'
import Pending from 'components/shared/Pending'
import { generateSkipText, generateEndGameText } from 'utils/helpers'

class PreRound extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			displayGameRules: false,
			timeRemaining: 15,
		}
	}

	componentDidMount() {
		//If it is the very beginning of the game, display the game information for all users for a few seconds.
		if (this.props.round === 1 && this.props.half === 'top') {
			this.setState(
				{
					displayGameRules: true,
				},
				() => {
					this.intervalId = setInterval(() => {
						this.decrementTimer()
					}, 1000)
				}
			)
		}
	}

	componentWillUnmount() {
		if (this.intervalId) {
			clearInterval(this.intervalId)
		}
	}

	decrementTimer = () => {
		if (this.state.timeRemaining > 0) {
			this.setState({
				timeRemaining: this.state.timeRemaining - 1,
			})
		} else {
			clearInterval(this.intervalId)
			this.setState({
				displayGameRules: false,
			})
		}
	}

	render() {
		const { teamScores, skipPenalty, endValue, ...rest } = this.props
		const scoresArr = [teamScores.team1, teamScores.team2]
		const { timeRemaining } = this.state
		const gameRulesText = timeRemaining > 7 ? generateSkipText(skipPenalty) : generateEndGameText('turns', endValue)
		const pendingMsg = timeRemaining > 7 ? 'shuffling cards' : 'preparing the timer'
		return this.state.displayGameRules ? (
			<TabooCard tabooWord="Game Rules">
				<InstructionsText>{gameRulesText}</InstructionsText>
				<Pending message={pendingMsg} speed={400}/>
			</TabooCard>
		) : (
			<React.Fragment>
				<ScoreCard teamScores={scoresArr} />
				<RoundInstructionsCard {...rest} />
			</React.Fragment>
		)
	}
}

PreRound.propTypes = {
	teamScores: PropTypes.object.isRequired,
	currentPlayer: PropTypes.object.isRequired,
	role: PropTypes.string.isRequired,
	giver: PropTypes.object.isRequired,
	watcher: PropTypes.object.isRequired,
	startRound: PropTypes.func.isRequired,
	error: PropTypes.string,
}

export default PreRound
