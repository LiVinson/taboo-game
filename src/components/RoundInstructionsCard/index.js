import RoundInstructionsCard from "./RoundInstructionsCard"

export default RoundInstructionsCard