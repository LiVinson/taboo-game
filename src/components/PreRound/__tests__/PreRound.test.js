import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import PreRound from '../PreRound'
import RoundInstructionsCard from 'components/RoundInstructionsCard'
import ScoreCard from 'components/ScoreCard'
import InstructionsText from 'components/shared/InstructionsText'

describe('PreRound renders and functions correctly', () => {
	const defaultProps = {
		teamScores: {
			team1: 3,
			team2: 5,
		},
		currentPlayer: {
			name: 'Sam',
			playerId: '12345',
		},
		round: 1,
		half: 'bottom',
		skipPenalty: 'half',
		endGameMethod: 'turns',
		endValue: 2,
		role: 'watcherTeam',
		giver: { name: 'Danielle', playerId: '12345', team: 'team 1' },
		watcher: { name: 'Stephen', playerId: '34567', team: 'team 1' },
		startRound: jest.fn(),
		error: null,
	}
	jest.useFakeTimers()
	test('PreRound renders correctly', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<PreRound {...props} />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})

	test('PreRound renders a ScoreCard and RoundInstructionsCard when not first round and first half', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<PreRound {...props} />)

		expect(wrapper.find(ScoreCard)).toHaveLength(1)
		expect(wrapper.find(RoundInstructionsCard)).toHaveLength(1)
	})

	test('PreRound renders InstructionsText for 15 seconds during first half of first round', () => {
		const props = {
			...defaultProps,
			half: 'top',
		}
		jest.useFakeTimers()
		const wrapper = mount(
			<ThemeProvider theme={theme}>
				<PreRound {...props} />
			</ThemeProvider>
		)
		const component = wrapper.find(PreRound)

		expect(component.find(InstructionsText)).toHaveLength(1)
		expect(component.find(RoundInstructionsCard)).toHaveLength(0)
		expect(component.find(ScoreCard)).toHaveLength(0)

		jest.advanceTimersByTime(16000)

		expect(component.contains(RoundInstructionsCard)).toBe(true)
		expect(component.contains(ScoreCard)).toBe(true)
	})
})
