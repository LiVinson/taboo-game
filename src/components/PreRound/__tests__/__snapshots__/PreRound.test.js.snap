// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreRound renders and functions correctly PreRound renders correctly 1`] = `
Array [
  <div
    className="sc-fzqNJr lpgMoX"
  >
    <p
      className="sc-fzoyAV TIEGc"
    >
      Scores
    </p>
    <ul
      className="sc-Axmtr dHjpTi"
    >
      <li
        className="sc-fzoLag ebyzUM"
      >
        Team 
        1
        : 
        <span>
          3
        </span>
      </li>
      <li
        className="sc-fzoLag ebyzUM"
      >
        Team 
        2
        : 
        <span>
          5
        </span>
      </li>
    </ul>
  </div>,
  <div
    className="sc-AxjAm hDlFTS"
  >
    <div
      className="sc-AxiKw dGnLMo"
    >
      Just Relax!
    </div>
    <div
      className="sc-AxhCb JZaCe"
    >
      <p
        className="sc-fzokOt kxxFRs"
      >
        It's the other team's turn to guess! 
        <span
          className="sc-fzqBZW hgCjBU"
        >
          Danielle
        </span>
         will be giving the clues, and
         
        <span
          className="sc-fzqBZW hgCjBU"
        >
          Stephen
        </span>
         will be checking for any illegal clues.
      </p>
    </div>
  </div>,
]
`;
